MoviesExample/app/src/main/java/com/novoda/demo/movies/MainActivity.java

```diff
@@ -30,7 +31,8 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
         setTitle("Top Rated Movies");
 
-        movieService = ((MoviesApplication) getApplication()).movieService();
+        MovieService movieService = ((MoviesApplication) getApplication()).movieService();
+        moviesViewModel = ViewModelProviders.of(this, new MoviesViewModelFactory(movieService)).get(MoviesViewModel.class);
```

Comment:
`ViewModelProviders` is an utility class provided by the lifecycle library to provision ViewModels. It ensures that ViewModels are reused when an Activity is recreated.
In this example, `MoviesViewModel` needs to be injected with `MovieService` in it's constructor. In order to do that, we need to provide a  `ViewModelFactory`. This class knows how to create our `MoviesViewModel` and it's used by `ViewModelProviders` the first time it needs to create a `MoviesViewModel`